### PR TITLE
Coerce componentconfig/v1alpha1 KubeSchedulerConfiguration to kubescheduler.config.k8s.io/v1alpha1

### PR DESCRIPTION
**Pull request by liggitt**
Since kube-scheduler [advised users to use a config file in previous releases](https://github.com/kubernetes/kubernetes/blob/v1.11.0/cmd/kube-scheduler/app/options/options.go#L147-L149), and warned that CLI flags were deprecated, it seems like we should ease the transition to the alpha apiVersion in 1.12, despite the previous version being alpha.

xref https://github.com/kubernetes/kubernetes/pull/66916#issuecomment-421148785

this now prints the following if you start the scheduler with an unsupported componentconfig/v1alpha1 config file:

> WARNING: the provided config file is an unsupported apiVersion ("componentconfig/v1alpha1"), which will be removed in future releases
> 
> WARNING: switch to command-line flags or update your config file apiVersion to "kubescheduler.config.k8s.io/v1alpha1"
> 
> WARNING: apiVersions at alpha-level are not guaranteed to be supported in future releases

/sig scheduling
/milestone v1.12

```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
cc @kubernetes/sig-scheduling-pr-reviews @luxas @dixudx @sttts

**Comment by liggitt** [comment]
/kind bug
(since this is mitigating the fallout of a previous bug)

File: cmd/kube-scheduler/app/options/configfile.go [hunk 1]
```diff
@@ -40,8 +45,48 @@ func loadConfigFromFile(file string) (*kubeschedulerconfig.KubeSchedulerConfigur
...
+			)
+			// attempt to decode to an unstructured object
+			obj, gvk, err := codec.Decode(data, nil, u)
+
+			// if this errored, or the object we read was not the legacy alpha gvk, return the original error
+			if err != nil || gvk == nil || *gvk != legacyConfigGVK {
+				return nil, originalErr
+			}
+
+			fmt.Printf("WARNING: config file is an unsupported alpha apiVersion=%q\n", legacyConfigGVK.GroupVersion().String())
```
Comment (dixudx) [inline cmd/kube-scheduler/app/options/configfile.go]:
```
fmt.Printf("WARNING: config file is an unsupported alpha apiVersion=%q, which will be deprecated in future version\n", legacyConfigGVK.GroupVersion().String())
```

**Review by dixudx — APPROVED** [review]
@liggitt Thanks for picking this up.

File: cmd/kube-scheduler/app/options/configfile.go (hunk 1 above)
Comment (liggitt) [reply cmd/kube-scheduler/app/options/configfile.go]: updated

**Review by yastij — APPROVED** [review]
/lgtm

**Comment by dims** [comment]
@bsalamat @k82cn do we need this before we cut 1.12? it's lacking approvals

**Comment by liggitt** [comment]
Yes, I think this is needed

**Review by timothysc — APPROVED** [review]
Not really a fan of how this is done, could you please open an issue to fix/remove the block in 1.13

/approve.

**Comment by liggitt** [comment]
> Not really a fan of how this is done, could you please open an issue to fix/remove the block in 1.13

Sure, will remove it myself as soon as master opens. I prefer keeping the violence localized.

**Comment by liggitt** [comment]
/approve. confused the bo
adding label manually

**Comment by liggitt** [comment]
> Not really a fan of how this is done, could you please open an issue to fix/remove the block in 1.13

as promised - https://github.com/kubernetes/kubernetes/pull/70040

**Pull request merged**